modules/social_features/social_group/modules/social_group_request/social_group_request.module

```diff
@@ -0,0 +1,257 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for social_group_request module.
+ */
+
+use Drupal\Core\Cache\RefinableCacheableDependencyInterface;
+use Drupal\Core\Url;
+use Drupal\group\Entity\GroupInterface;
+use Drupal\social_group_request\Plugin\GroupContentEnabler\GroupMembershipRequest;
+use Drupal\group\Entity\GroupContentInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function social_group_request_group_content_insert(GroupContentInterface $group_content) {
+  $type = $group_content->getContentPlugin()->getPluginId();
+  $membership_request_enabled = $group_content->getGroup()->getGroupType()->hasContentPlugin('group_membership_request');
+
+  // If new membership is created, check related membership request exists.
+  if ($type === 'group_membership' && $membership_request_enabled) {
+    $approver = \Drupal::currentUser()->id();
+    $storage = \Drupal::entityTypeManager()->getStorage('group_content');
+    $filters = [
+      'entity_id' => $group_content->getEntity()->id(),
+      'grequest_status' => GroupMembershipRequest::REQUEST_PENDING,
+    ];
+    $requests = $storage->loadByGroup($group_content->getGroup(), 'group_membership_request', $filters);
+
+    /** @var \Drupal\group\Entity\GroupContentInterface $request */
+    foreach ($requests as $request) {
+      $request
+        ->set('grequest_status', GroupMembershipRequest::REQUEST_ACCEPTED)
+        ->set('grequest_updated_by', $approver)
+        ->save();
+
+      _activity_basics_entity_action($request, 'change_status_group_membership_request_entity_action');
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_group().
+ */
+function social_group_request_preprocess_group__hero(&$variables) {
+  /** @var \Drupal\group\Entity\GroupInterface $group */
+  $group = $variables['group'];
+  $account = \Drupal::currentUser();
+
+  if ($group->getGroupType()->id() !== 'closed_group') {
```

Comment:
Why limit yourself already to closed_group. 
I know there is a follow up but `$group->getGroup()->getGroupType()->hasContentPlugin('group_membership_request');`  would already make our lives much easier in the future and its the same code you used a couple of lines before